ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -136,7 +137,23 @@ ompi_mtl_ofi_component_register(void)
                                     MCA_BASE_VAR_SCOPE_READONLY,
                                     &ompi_mtl_ofi.ofi_progress_event_count);
 
-     free(desc);
+    free(desc);
+
+    fallback_alternative_tag = false;
+    asprintf(&desc, "Use alternative ofi tag bits distribution for providers that do not support FI_REMOTE_CQ_DATA:"
```

Comment:
I don't love this.  I've read it three times and I'm still confused.  I also don't like that there's no way to force using cq_data (or fail if it's not there).  Maybe a 3-option flag that sets a particular mode (would be easier to test as well, wouldn't need a device that doesn't support cq_data to test, could just run through each mode).

Reply:
There are few different things here: 
 - The README file was intended to provider further clarity to the fallback tag thing. I'm open to add any additional mechanism to add more clarity, please share how. 
-  I understand the complications for testing, but I'm not convinced that  CQ_DATA selection should be made available through an mca parameter since this is just a testing requirement with no benefit for the end user (in fact it limits scalability). Unless there is a provider that may have benefits by intentionally not using CQ_DATA, e.g. overhead in supporting FI_DIRECTED_RECIEVE, required to not include the rank in the tag.  Thoughts ?